dev/releases/utils.py

```diff
@@ -128,29 +127,34 @@ def safe_git_fetch_tags():
         error('failed to fetch tags, you may have to do \n'
               + 'git fetch --tags -f')
 
-
-# Returns a boolean
-def check_whether_git_tag_exists(tag):
-    safe_git_fetch_tags()
-    res = subprocess.run(["git", "tag", "-l"],
-                         capture_output=True,
-                         text=True,
-                         check=True)
-    tags = res.stdout.split('\n')
-    for s in tags:
-        if tag == s:
-            return True
-    return False
-
-# Returns a boolean
-def check_whether_github_release_exists(tag):
-    if CURRENT_REPO == None:
-        print("CURRENT_REPO is not initialized. Call initialize_github first")
-    releases = CURRENT_REPO.get_releases()
-    for release in releases:
-        if release.tag_name == tag:
-            return True
-    return False
+# lightweight vs annotated
+# https://stackoverflow.com/questions/40479712/how-can-i-tell-if-a-given-git-tag-is-annotated-or-lightweight#40499437
+def is_annotated_git_tag(tag):
+    res = subprocess.run(["git", "for-each-ref", "refs/tags/" + tag],
+                         capture_output=True, text=True)
+    return res.returncode == 0 and res.stdout.split()[1] == "tag"
+
+def get_commit_of_git_tag(tag):
+    res = subprocess.run(["git", "show-ref", "--tags", "--dereference", tag],
+                            check=True, capture_output=True, text=True)
+    res = res.stdout.split('\n')
+    res = [obj for obj in res if obj.find(tag + "^{}") != -1]
+    if len(res) == 0:
+        error(f"Could not find tag {tag}")
+    res = res[0]
+    return res.split()[0]
```

Comment:
This is needlessly complicated, though: you could just do `git rev-parse v4.11.1^{}` -- and you already do that below for `HEAD`

Reply:
I added your suggestions, thanks!